components/view_image/view_image.jsx

```diff
@@ -240,11 +263,13 @@ export default class ViewImageModal extends React.PureComponent {
                     />
                 );
             } else if (fileInfo && fileInfo.extension && fileInfo.extension === FileTypes.PDF) {
+                this.setState({showZoomControls: true});
```

Comment:
setting setState in render is not a good idea. Can we change the setSate to be on `getDerivedStateFromProps` or any other lifecycle you see fit?

Reply:
Having some trouble with this one. More I think about it, I'm not sure if it's even worth making `showZoomControls` a part of the state? My understanding is when `setState` is called, `render()` is called which seems unnecessary. All I'm trying to do is set a variable to true if it's a PDF so the popup bar will display the zoom controls.

Reply:
I think I figured it out. `getDerivedStatesFromProps` seems like the best place to put it as far as I can tell.